diff --git a/node_modules/@astrojs/starlight/.prettierignore b/node_modules/@astrojs/starlight/.prettierignore
new file mode 100644
index 0000000..72e8ffc
--- /dev/null
+++ b/node_modules/@astrojs/starlight/.prettierignore
@@ -0,0 +1 @@
+*
diff --git a/node_modules/@astrojs/starlight/components/SidebarSublist.astro b/node_modules/@astrojs/starlight/components/SidebarSublist.astro
index 399d191..b37ecdb 100644
--- a/node_modules/@astrojs/starlight/components/SidebarSublist.astro
+++ b/node_modules/@astrojs/starlight/components/SidebarSublist.astro
@@ -2,6 +2,15 @@
 import { flattenSidebar, type SidebarEntry } from '../utils/navigation';
 import Icon from '../user-components/Icon.astro';
 import Badge from './Badge.astro';
+import { Icons } from '@astrojs/starlight/components/Icons';
+import VscodeIconsFileTypeJsOfficial from '~icons/vscode-icons/file-type-js-official'
+import VscodeIconsFileTypeCss from '~icons/vscode-icons/file-type-css'
+import VscodeIconsFileTypeHtml from '~icons/vscode-icons/file-type-html'
+import LogosTypescriptIcon from '~icons/logos/typescript-icon'
+import CarbonHttp from '~icons/carbon/http'
+import VscodeIconsFileTypeLightPrettier from '~icons/vscode-icons/file-type-light-prettier'
+import LogosNextjsIcon from '~icons/logos/nextjs-icon'
+import VscodeIconsFileTypeEslint from '~icons/vscode-icons/file-type-eslint'
 
 interface Props {
 	sublist: SidebarEntry[];
@@ -39,6 +48,23 @@ const { sublist, nested } = Astro.props;
 					>
 						<summary>
 							<div class="group-label">
+                {entry.icon && entry.icon === 'http'
+                  ? <CarbonHttp style={{fontSize: '1.2rem'}} />
+                  : entry.icon === 'javascript'
+                  ? <VscodeIconsFileTypeJsOfficial />
+                  : entry.icon === 'typescript'
+                  ? <LogosTypescriptIcon />
+                  : entry.icon === 'css'
+                  ? <VscodeIconsFileTypeCss />
+                  : entry.icon === 'html'
+                  ? <VscodeIconsFileTypeHtml />
+                  : entry.icon === 'prettier'
+                  ? <VscodeIconsFileTypeLightPrettier />
+                  : entry.icon === 'nextjs'
+                  ? <LogosNextjsIcon  style={{fontSize: '1.0rem'}} />
+                  : entry.icon === 'eslint'
+                  ? <VscodeIconsFileTypeEslint />
+                  : <Icon name={'seti:' + entry.icon as keyof typeof Icons} size=".8rem" />}
 								<span class="large">{entry.label}</span>
 								{entry.badge && (
 									<>
diff --git a/node_modules/@astrojs/starlight/schemas/sidebar.ts b/node_modules/@astrojs/starlight/schemas/sidebar.ts
index 86eb7f8..53425e0 100644
--- a/node_modules/@astrojs/starlight/schemas/sidebar.ts
+++ b/node_modules/@astrojs/starlight/schemas/sidebar.ts
@@ -50,6 +50,7 @@ const AutoSidebarGroupSchema = SidebarGroupSchema.extend({
 		/** How many directories deep to include from this directory in the sidebar. Default: `Infinity`. */
 		// depth: z.number().optional(),
 	}),
+  icon: z.string().optional(),
 });
 export type AutoSidebarGroup = z.infer<typeof AutoSidebarGroupSchema>;
 
diff --git a/node_modules/@astrojs/starlight/utils/navigation.ts b/node_modules/@astrojs/starlight/utils/navigation.ts
index 7be5253..9c41ff1 100644
--- a/node_modules/@astrojs/starlight/utils/navigation.ts
+++ b/node_modules/@astrojs/starlight/utils/navigation.ts
@@ -32,6 +32,7 @@ interface Group {
 	entries: (Link | Group)[];
 	collapsed: boolean;
 	badge: Badge | undefined;
+  icon?: string;
 }
 
 export type SidebarEntry = Link | Group;
@@ -107,6 +108,7 @@ function groupFromAutogenerateConfig(
 		entries: sidebarFromDir(tree, currentPathname, locale, subgroupCollapsed ?? item.collapsed),
 		collapsed: item.collapsed,
 		badge: item.badge,
+    icon: item.icon,
 	};
 }
 
